### PR TITLE
feat: variables

### LINKED ISSUE #1: feat: Cairo Variables
This example will introduce the three types of variables commonly found in Cairo contracts:
- Local variables, who are declared inside a function and not stored on the blockchain
- Storage variables, declared in the `Storage` struct, stored on the blockchain and that can be accessed from one execution to another
- global variables, who provide information about the blockchain and are accessed with library functions: `get_caller_address`, `get_contract_address`,  etc...

Solidity reference: https://solidity-by-example.org/variables/

### PR DESCRIPTION
**Pull request by julio4**
Close #1 
@enitrat Can you please review this and let me know if the explanations and examples are what you expected?

### CODE REVIEW

File: src/variables/global_variables.cairo [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#[contract]
+mod GlobalExample {
+    // import the library function and types from the starknet core library 
+    use starknet::get_caller_address;
+    use starknet::ContractAddress;
+
+    #[external]
+    fn transfer(recipient: ContractAddress, amount: u256) {
+        // Call the get_caller_address function to get the sender address
```
Comment (enitrat) [inline src/variables/global_variables.cairo:9]: > to get the sender address

let's call it `caller` address

File: src/variables/global_variables.cairo [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#[contract]
+mod GlobalExample {
+    // import the library function and types from the starknet core library 
+    use starknet::get_caller_address;
+    use starknet::ContractAddress;
+
+    #[external]
+    fn transfer(recipient: ContractAddress, amount: u256) {
```
Comment (enitrat) [inline src/variables/global_variables.cairo:8]: If your function doesn't perform a transfer, you shouldn't call it this way. You can give it a generic name like `foo` or `do_something`. Also, we don't need any parameters in this example.

File: src/variables/local_variables.cairo [hunk 3]
```diff
@@ -0,0 +1,14 @@
+#[contract]
+mod LocalVariables {
+    #[view]
+    fn doSomething(value: u32) -> u32 {
```
Comment (enitrat) [inline src/variables/local_variables.cairo:4]: in Cairo, we use snake_case for functions!

**Review by enitrat — APPROVED** [review]
Looks good! Please fix the small comments and then we can merge :)

**Comment by julio4** [comment]
For global variables/functions, should we add a list or a link to more detailed references?

**Comment by enitrat** [comment]
> For global variables/functions, should we add a list or a link to more detailed references?

To do this we would need to have a proper documentation of the corelib, which is not the case yet. Let's keep it this way for now.

**Pull request merged**